packages/react-data-grid/src/ReactDataGrid.tsx

```diff
@@ -0,0 +1,729 @@
+      const SelectAllRenderer = <SelectAllComponent onChange={this.handleCheckboxChange} ref={this.selectAllCheckbox} />;
+      const headerRenderer = props.enableRowSelect === 'single' ? null : SelectAllRenderer;
+      const Formatter = (this.props.rowActionsCell ? this.props.rowActionsCell : CheckboxEditor) as unknown as React.ComponentClass<{ rowSelection: unknown }>;
+      const selectColumn = {
+        key: 'select-row',
+        name: '',
+        formatter: <Formatter rowSelection={this.props.rowSelection} />,
+        onCellChange: this.handleRowSelect,
+        filterable: false,
+        headerRenderer,
+        width: 60,
+        frozen: true,
+        getRowMetaData: (rowData: RowData) => rowData,
+        cellClass: this.props.rowActionsCell ? 'rdg-row-actions-cell' : ''
+      };
+      cols.unshift(selectColumn as unknown as Column);
+    }
+    this._cachedComputedColumns = cols;
+
+    return this._cachedComputedColumns;
+  }
+
+  render() {
+    const cellMetaData: CellMetaData = {
+      rowKey: this.props.rowKey,
+      onCellClick: this.handlerCellClick,
+      onCellContextMenu: this.handleCellContextMenu,
+      onCellDoubleClick: this.handleCellDoubleClick,
+      onCellExpand: this.props.onCellExpand,
+      onRowExpandToggle: this.props.onRowExpandToggle,
+      getCellActions: this.props.getCellActions,
+      onDeleteSubRow: this.props.onDeleteSubRow,
+      onAddSubRow: this.props.onAddSubRow,
+      onDragEnter: this.handleDragEnter
+    };
+    if (this.props.cellRangeSelection) {
+      cellMetaData.onCellMouseDown = this.handleCellMouseDown;
+      cellMetaData.onCellMouseEnter = this.handleCellMouseEnter;
+    }
+
+    const interactionMasksMetaData: InteractionMasksMetaData = {
+      onCheckCellIsEditable: this.props.onCheckCellIsEditable,
+      onCellCopyPaste: this.props.onCellCopyPaste,
+      onGridRowsUpdated: this.handleGridRowsUpdated,
+      onDragHandleDoubleClick: this.handleDragHandleDoubleClick,
+      onCellSelected: this.props.onCellSelected,
+      onCellDeSelected: this.props.onCellDeSelected,
+      onCellRangeSelectionStarted: this.props.cellRangeSelection && this.props.cellRangeSelection.onStart,
+      onCellRangeSelectionUpdated: this.props.cellRangeSelection && this.props.cellRangeSelection.onUpdate,
+      onCellRangeSelectionCompleted: this.props.cellRangeSelection && this.props.cellRangeSelection.onComplete,
+      onCommit: this.handleCommit
+    };
+
+    let containerWidth: string | number = this.props.minWidth || this.gridWidth();
+    let gridWidth: string | number = containerWidth;
+
+    // depending on the current lifecycle stage, gridWidth() may not initialize correctly
+    // this also handles cases where it always returns undefined -- such as when inside a div with display:none
+    // eg Bootstrap tabs and collapses
+    if (typeof containerWidth === 'undefined' || isNaN(containerWidth) || containerWidth === 0) {
```

Comment:
If we're using `isNaN`, then the typeof undefined check is redundant.
Maybe we want to use `Number.isNaN` instead?

Reply:
I do not think either `isNaN` or `undefined` check is required as the result is always a number. I am keeping `isNaN` check just in case